### ISSUE TITLE
Training Process Problems

### ISSUE THREAD

**Issue by zomkey** [microsoft/Deep3DFaceReconstruction#51]
Hi，thanks for your work, I am really interested !
Now I have implement training process according to your paper, but I found some problems during training:
1. Image-level loss is hard to converge，with which the loss is about 40~50 at last , leading the reconstruction result to be mean shape without real texture of input image. (I used skin mask to train, and I want to ask whether there exists any tricks to converge Image-level loss ?)
2. Land-mark loss and image-level loss is hard to balance, which means that if I increase the weight of Land-mark loss, which will lead reconstruction result to have accurate pose and expression but wrong face color. On the contrast, if the weight of image-level loss is increased, the reconstruction result can have accurate face color but wrong pose and expression. (I want to ask How to balance this two losses?)
3. The learning rate is also important during training, which Learning rate strategy is used to achieve good results ?
Hope for your advice, thank you very much!

**Comment by YuDeng** [comment]
Hi, thanks for your interest in this work. For photo-level loss, we normalize the color to 0-1 for output images and as a reference, the training loss drops to 0.1 in my training process. You could also try to reduce the weight of regularization on textures.
The ratio between the weighted value (loss*weight) of photo-loss and landmark-loss is around 10 in my case. 
We do not carefully adjust the learning rate. We only use a Adam optimize with constant learning rate throughout the experiment.

**Comment by zomkey** [comment]
Thanks for your advice, now I can achieve a convergency result.

**Comment by hangon666** [comment]
@zomkey 
Hi,  I am very interested in this work too,   if convenient,  could you send me your training codes so that I can learn the process better?
My email address is : cherry_2lin@126.com
Thank you very much!

**Comment by Moon-W** [comment]
@zomkey 
Hello, can you share your training code with me? My email : ChrisandPaul3@163.com
Thanks a lot

**Comment by Chen-Jinlong** [comment]
> @zomkey
> Hi, I am very interested in this work too, if convenient, could you send me your training codes so that I can learn the process better?
> My email address is : [cherry_2lin@126.com](mailto:cherry_2lin@126.com)
> Thank you very much!

Hi~Did you get the training codes? Recently, I am working on this project, too, so I want to know if you have got the training codes since I need that for my thesis.

**Comment by xingmimfl** [comment]
> Thanks for your advice, now I can achieve a convergency result.

I also always get mean shape. have you solved this problem??

**Comment by HOMGH** [comment]
> @zomkey
> Hi, I am very interested in this work too, if convenient, could you send me your training codes so that I can learn the process better?
> My email address is : [cherry_2lin@126.com](mailto:cherry_2lin@126.com)
> Thank you very much!

Hi,
Did you get the training code? Would you please sent it to me?

**Comment by ZMpursue** [comment]
@zomkey 
Wow, I am very interested in this work too, if convenient, could you send me your training codes so that I can learn the process better?
My email address is :qq1063653217@outlook.com
Thank you!

**Comment by UestcJay** [comment]
> Thanks for your advice, now I can achieve a convergency result.

Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 824218055@qq.com

**Comment by JacksonL1** [comment]
> Thanks for your advice, now I can achieve a convergency result.

Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 824218055@qq.com

**Comment by JacksonL1** [comment]
> Thanks for your advice, now I can achieve a convergency result.

Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 276810962@qq.com

**Comment by DouYishun** [comment]
> Thanks for your advice, now I can achieve a convergency result.

Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 10386822@qq.com

**Comment by mingsjtu** [comment]
Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 245511342@qq.com

**Comment by lhh753159** [comment]
@zomkey
Hi, I am very interested in this work too, if convenient, could you send me your training codes ? Thank you very much!
My email address is : 984146874@qq.com
